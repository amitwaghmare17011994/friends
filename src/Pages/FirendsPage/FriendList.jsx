import React from 'react'

function FriendList(props) {
    const { friends = [] } = props;
    return (
        <>
            {
                friends.map((friendItem) => {
                    return <div key={friendItem.fullName} className={'friend_item'}>{friendItem.fullName}</div>
                })
            }

        </>
    )
}

export default FriendList
