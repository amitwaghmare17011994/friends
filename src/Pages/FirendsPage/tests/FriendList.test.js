import { create } from 'react-test-renderer';
import FriendList from "../FriendList"

describe('FriendList tests', () => {
    function getRoot(props = {}) {
        return create(<FriendList {...props} />).root

    }

    it('render firend list', () => {
        const root = create(<FriendList />)
        expect(root.toJSON()).toMatchSnapshot();
    })

    it('render 2 items with className friendItem', () => {
        const friends = [{ fullName: 'amit' }, { fullName: 'sumit' }]
        const root = getRoot({friends})

        const allFreindItems = root.findAllByProps({ className: 'friend_item' })
        expect(allFreindItems.length).toBe(friends.length)
    })
 
})