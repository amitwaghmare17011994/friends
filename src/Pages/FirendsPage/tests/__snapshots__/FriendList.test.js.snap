// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FriendList tests render firend list 1`] = `null`;
